CONTRIBUTING.md

```diff
@@ -0,0 +1,60 @@
+Contributing
+============
+Help is very welcomed. Code contributions must be done in respect of [PSR-2](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md).
+More details on how to contribute and guidelines for [pull requests](http://elastica.io/contribute/pull-request.html) can be found [here](http://elastica.io/contribute/).
+
+See [Coding guidelines](http://elastica.io/contribute/coding-guidelines.html) for tips on how to do so.
+All changes which are made to the project are added to the [CHANGELOG.md](https://github.com/ruflin/Elastica/blob/master/CHANGELOG.md).
+
+
+Issues
+------
+* Bugs & Feature requests: If you found a bug, open an issue on [Github](https://github.com/ruflin/Elastica/issues). Before you open it search the issues which were already reported to make sure the issue doesn't already exist. Make sure to report the issue in a way that it can be reproduced and you report which version of Elastica and its dependencies you are using. Best is to report the problem with a code example.
+* Questions & Problems: If you have questions or a specific problem, use the [Elastica Gitter](https://gitter.im/ruflin/Elastica) Chat or open an issue on [Stackoverflow](http://stackoverflow.com/questions/tagged/elastica). Make sure to assign the tag Elastica.
+
+
+Setup
+-----
+Elastica currently allows two setups for development. Either through vagrant or docker-compose. To use the vagrant environment, run `vagrant up`. To use the docker environment, check out the Makefile for the necessary commands.
+* Run your changes / tests in the virtual environment to make sure it is reproducible. Don't use a local setup
+* Run tests locally on your machine first
```

Comment:
Maybe a bit confusing with that "Don't use a local setup" before.

Better:

* Run your changes / tests in the virtual environment to make sure it is reproducible.
* Run the tests before creating the pull request using vagrant or docker-compose locally.